File: frontend/src/services/recoverPasswordService.ts
import type { CreateRecoverPasswordRequest, RecoverPasswordRequest } from "@/types/api/recoverPassword";
import type { DefaultResponse } from "@/types/api/default";
import { handleApiError } from "@/utils/handleApiError";
import { axios } from "@/api/axios";

const createRecoverPassword = async (data: CreateRecoverPasswordRequest): Promise<DefaultResponse<null>> => {
  try {
    const response = await axios.post("/recover-password", data);
    return response.data;
  } catch (error) {
    handleApiError(error, "Erro ao realizar a criação de recuperação de senha");
  }
};

const recoverPassword = async ({ data, recoverToken }: { recoverToken: string; data: RecoverPasswordRequest }): Promise<DefaultResponse<null>> => {
  try {
    const response = await axios.put(`/recover-password/${recoverToken}`, data);
    return response.data;
  } catch (error) {
    handleApiError(error, "Erro ao realizar a recuperar a senha");
  }
};

export const recoverPasswordService = {
  createRecoverPassword,
  recoverPassword,
};

File: backend/src/main/java/com/guisebastiao/authentication/service/RecoverPasswordService.java
package com.guisebastiao.authentication.service;

import com.guisebastiao.authentication.dto.DefaultResponse;
import com.guisebastiao.authentication.dto.request.CreateRecoverPasswordRequest;
import com.guisebastiao.authentication.dto.request.RecoverPasswordRequest;

public interface RecoverPasswordService {
    DefaultResponse<Void> createRecoverPassword(CreateRecoverPasswordRequest dto);
    DefaultResponse<Void> resetPassword(String recoverToken, RecoverPasswordRequest dto);
}
